_posts/2017-07-09-snowplow-r90-lascaux-released.md

```diff
@@ -0,0 +1,148 @@
+tags: [snowplow, shred, relational databases]
+author: Anton
+category: Releases
+---
+
+We are tremendously excited to announce the release of [Snowplow 90 Lascaux][snowplow-release].
+This release introduces RDB Loader, a new EMR application replacing StorageLoader, according to our [Splitting EmrEtlRunner RFC][splitting-eer-rfc] as well as various enhancements in EmrEtlRunner.
+
+Read on for more information on R90 Lascaux, named after [Upper Paleolithic cave complex in southwestern France][lascaux]:
+
+1. [Introducing RDB Loader](/blog/2017/07/07/snowplow-r90-lascaux-released-moving-loading-step-onto-emr#rdb-loader)
+2. [Other changes](/blog/2017/07/07/snowplow-r90-lascaux-released-moving-loading-step-onto-emr#other)
+5. [Upgrading](/blog/2017/07/07/snowplow-r90-lascaux-released-moving-loading-step-onto-emr#upgrading)
+
+![lascaux][lascaux-img]
+
+<!--more-->
+
+<h2 id="rdb-loader">1. Introducing RDB Loader</h2>
+
+StorageLoader was standalone JRuby app, running along with EmrEtlRunner on EC2 box and allowing our users to ingest shredded data into several relational databases, such as AWS Redshift or PostgreSQL. Approach with standalone application served us quite well since Snowplow batch pipeline inception. However, as we're moving towards supporting [new cloud providers][azure-rfc] and [simplifying existing orchestration tools][splitting-eer-rfc], we decided to modularize our
+
+* StorageLoader contains too much logic that can and must be executed by generic tools such as Dataflow Runner or SQL Runner
+* StorageLoader contains too much logic that can and must be executed by cloud-specific tools, such as `S3DistCp` or Elastic MapReduce itself.
+* StorageLoader is essential part of data pipelines, where end storage is relational database, whereas many possible targets can omit shredding step and work well with enriched data
+
+RDB Loader is feature-identical replacement for StorageLoader.
+
+Decouple data-ingestion and data-movement
+
+<h2 id="other">2. Other changes</h2>
+
+Along with shifting from standalone app to EMR step we also made several important improvements in:
+
+* Whole codebase now written in Scala [#3023][issue-3023]
+* RDB Loader loads all existing run folders from shredded good folder [#2962][issue-2962]
+* Fixed eventual consistency problem [#3113][issue-3113]
+
+Additionally, improving on [R89][r89], we're introducing a way to specify arbitrary EMR
+configuration options through the EmrEtlRunner configuration file like so:
+
+{% highlight yaml %}
+aws:
+  emr:
+    // ...
+    configuration:
+      yarn-site:
+         yarn.resourcemanager.am.max-attempts: "1"
+       spark:
+         maximizeResourceAllocation: "true"
+       spark-defaults:
+         spark.executor.instances: "17"
+         spark.yarn.executor.memoryOverhead: "4096"
+         spark.executor.memory: "35G"
+         spark.yarn.driver.memoryOverhead: "4096"
+        // etc
+{% endhighlight %}
+
```

Comment:
@BenFradet please add a little context on how this is useful - basically Spark needs more tuning/handholding than EMR, and we are now giving you the tools to do it; plus expect some guides on our Discourse from us and the wider community on how to make use of these custom config options, given your specific event workloads, in the future...